code42_connector.py

```diff
@@ -370,22 +458,85 @@ def _build_alerts_query(self, username, start_date, end_date, alert_state):
             filters.append(Actor.eq(username))
         if alert_state is not None:
             filters.append(AlertState.eq(alert_state))
-        filters.append(self._build_date_range_filter(start_date, end_date))
+        filters.append(
+            self._build_date_range_filter(DateObserved, start_date, end_date)
+        )
         query = AlertQuery.all(*filters)
         return query
 
-    def _build_date_range_filter(self, start_date, end_date):
+    def _build_file_events_query(
+        self,
+        start_date,
+        end_date,
+        file_hash,
+        file_name,
+        file_path,
+        file_category,
+        username,
+        hostname,
+        private_ip,
+        public_ip,
+        exposure_type,
+        process_name,
+        url,
+        window_title,
+        untrusted_only,
+    ):
+        filters = []
+        if file_hash:
+            if utils.is_md5(file_hash):
+                filters.append(MD5.eq(file_hash))
+            elif utils.is_sha256(file_hash):
+                filters.append(SHA256.eq(file_hash))
+            else:
+                raise Code42UnsupportedHashError()
+
+        if file_name:
+            add_eq_filter(filters, file_name, FileName)
+        if file_path:
+            add_eq_filter(filters, file_path, FilePath)
+        if file_category:
+            add_eq_filter(filters, file_category, FileCategory)
+        if hostname:
+            add_eq_filter(filters, hostname, OSHostname)
+        if username:
+            add_eq_filter(filters, username, DeviceUsername)
+        if private_ip:
+            add_eq_filter(filters, private_ip, PrivateIPAddress)
+        if public_ip:
+            add_eq_filter(filters, public_ip, PublicIPAddress)
+        if exposure_type:
+            if exposure_type == "All":
+                filters.add(ExposureType.exists)
+            else:
+                add_eq_filter(filters, exposure_type, ExposureType)
+        if process_name:
+            add_eq_filter(filters, process_name, ProcessName)
+        if url:
+            add_eq_filter(filters, url, TabURL)
+        if window_title:
+            add_eq_filter(filters, window_title, WindowTitle)
+        if untrusted_only:
+            filters.append(TrustedActivity.is_false())
+
+        filters.append(
+            self._build_date_range_filter(EventTimestamp, start_date, end_date)
+        )
+        query = FileEventQuery.all(*filters)
+        return query
+
+    def _build_date_range_filter(self, date_filter_cls, start_date, end_date):
         if start_date and not end_date:
-            return DateObserved.on_or_after(dateutil.parser.parse(start_date))
+            return date_filter_cls.on_or_after(dateutil.parser.parse(start_date))
```

Comment:
modified this func slightly to make it so that it could also be used with `EventTimestamp`